cmd/storagenode-updater/main.go

```diff
@@ -168,26 +168,26 @@ func update(ctx context.Context, binPath, serviceName string, renameBinary renam
 
 	client := checker.New(runCfg.ClientConfig)
 	log.Println("downloading versions from", runCfg.ServerAddress)
-	shouldUpdate, newVersion, err := client.ShouldUpdate(ctx, serviceName, nodeID)
+	processVersion, err := client.Process(ctx, serviceName)
 	if err != nil {
 		return errs.Wrap(err)
 	}
 
-	if shouldUpdate {
-		// TODO: consolidate semver.Version and version.SemVer
-		suggestedVersion, err := newVersion.SemVer()
-		if err != nil {
-			return errs.Wrap(err)
-		}
+	// TODO: consolidate semver.Version and version.SemVer
+	suggestedVersion, err := processVersion.Suggested.SemVer()
+	if err != nil {
+		return errs.Wrap(err)
+	}
 
-		if currentVersion.Compare(suggestedVersion) < 0 {
+	if currentVersion.Compare(suggestedVersion) < 0 {
+		if version.ShouldUpdate(processVersion.Rollout, nodeID) {
```

Comment:
Instead of indenting the main logic, can we reorganize the code like this:
```
if currentVersion.Compare(suggestedVersion) >= 0 {
	log.Printf("%s version is up to date\n", serviceName)
	return nil
}

if !version.ShouldUpdate(processVersion.Rollout, nodeID) {
	log.Printf("new %s version available but not rolled out to this nodeID yet\n", serviceName)
	return nil
}

// main logic
```
This makes it easier to read what we logic in each case.